darts/models/forecasting/pl_forecasting_module.py

```diff
@@ -226,12 +253,28 @@ def _compute_loss(self, output, target):
         if self.likelihood:
             return self.likelihood.compute_loss(output, target)
         else:
-            # If there's no likelihood, nr_params=1 and we need to squeeze out the
+            # If there's no likelihood, nr_params=1, and we need to squeeze out the
             # last dimension of model output, for properly computing the loss.
             return self.criterion(output.squeeze(dim=-1), target)
 
+    def _calculate_metrics(self, output, target, metrics):
+        if self.likelihood:
```

Comment:
let's check if there is a metric before computing anything that is not required (such as sampling)

```suggestion
        if not len(metrics):
            return
        
        if self.likelihood:
            ...
```